docs/logging-guidelines.md

```diff
@@ -0,0 +1,167 @@
+# Lightweight Golem Logging Guidelines
+
+## Objective
+
+The purpose of this article is to provide a set of prescriptive guidelines for developers to follow in order to achieve consistent content of execution logs.
+Execution logs serve two main purposes, depending on "audience":
+- **Core system developers** - Efficient troubleshooting and debugging during development of Golem modules.
+- **Integrator developers** - Troubleshooting and debugging of applications which use Golem as platform.
+- **Users** - Diagnostics of issues appearing on owned Golem nodes (eg. setup, infrastructural or maintenance-related issues).
+
+Ideal logs contain the right amount of information for the situation and audience. This implies that "too much logs" may be as useless as "no logs" - therefore focus is put on recording appropriate information at each log level.
+
+The article includes also **generic requirements** referring to the **logging framework** chosen by developers for specific platform they are working on - this takes into account that Golem is a multi-platform software ecosystem, and however many programming languages can (in theory) be used to develop Golem components, their developers should maintain consistent approach to logs & audit trails, to provide uniform level of "user & developer experience".
+
+## Scope
+
+These guidelines adhere to software published by Golem Factory as parts of Lightweight Golem/Yagna ecosystem. This includes among others, the following repos:
+- [yagna](https://github.com/golemfactory/yagna)
+- [ya-client](https://github.com/golemfactory/ya-client)
+- [yapapi](https://github.com/golemfactory/yapapi)
+- [yagna-integration](https://github.com/golemfactory/yagna-integration)
+
+## Logging library requirements
+
+Must have:
+- Redirect log stream to **console** (STDOUT/STDERR) and/or **files**
+- Filter log entries by log level
+- Filter log entries by entry subsets (like namespace subtrees)
+
+Nice to have:
+- ...
+
+## Log entry content
+
+The log entries should record following aspects/attributes:
+
+**Must have:**
+- **Timestamp** (with ms granularity, in UTC or with TZ information) - must be possible to correlate entries from nodes in different timezones
+- Log **level**
+- Log **area/topic** (eg. namespace or module of code which records the log entry)
+- **Grouping or correlating** attribute (attribute which allows to filter events related to a single command, activity, API request, etc.)
+  - Thread id
+  - Bespoke correlation id
+- Log entry **description** (human readable, no linebreaks)
```

Comment:
Sometimes "no linebreaks" is opposite of "human readable"

Reply:
Agreed, rewritten